### PR TITLE
Update pageToken to have opaque semantics in common/reducer.js

### LINKED ISSUE #2828: Rename pagingNum to pageToken in media/common/reducer.js
## Task Description

Use common terminology between 'local' and 'media3p' by renaming 'pagingNum' to 'pageToken'.
This change should extend to reducer.js and actions.js, with minimal changes to local/useContextValueProvider.js, which can continue to use the pagingNum terminology where appropriate.

### PR DESCRIPTION
**Pull request by cvolzke4**
## Summary

Update pageToken to have opaque semantics in common/reducer.js.

This removes the assumption that pageToken (formally pagingNum) is always an integer that represents a page number. This allows local and media3p pagination to behave in a consistent manner, for example:

local: `pageToken == <page number>`
coverr: `pageToken == <page number>`
unsplash: `pageToken == <next page url>`
tenor: `pageToken == <page token>`

## To-do

- Write tests for local/reducer.js and common/reducer.js (https://github.com/google/web-stories-wp/issues/2512).
- QA review, or add a feature flag, but this should not be necessary

## User-facing changes

No user-facing changes.

## Testing Instructions

- Open the media pane to verify that it appears.
- Scroll down to verify a page of results in returned.
- Scroll to the last page to make sure it loads correctly.
- Upload an image successfully.
- Upload a video successfully.
- Insert an image into the canvas.

### CODE REVIEW

**Review by diegovar — APPROVED** [review]

File: assets/src/edit-story/app/media/local/useContextValueProvider.js [hunk 1]
```diff
@@ -72,11 +72,13 @@ export default function useContextValueProvider(reducerState, reducerActions) {
         .then(({ data, headers }) => {
           const totalPages = parseInt(headers.get('X-WP-TotalPages'));
           const mediaArray = data.map(getResourceFromAttachment);
+          const hasMore = p < totalPages;
```
Comment (winterDroid) [inline assets/src/edit-story/app/media/local/useContextValueProvider.js:75]: this won't be supported with a token right?

File: assets/src/edit-story/app/media/local/useContextValueProvider.js [hunk 2]
```diff
@@ -72,11 +72,13 @@ export default function useContextValueProvider(reducerState, reducerActions) {
...
           callback({
             media: mediaArray,
             mediaType: currentMediaType,
             searchTerm: currentSearchTerm,
             pageToken: p,
+            nextPageToken: hasMore ? p + 1 : undefined,
```
Comment (winterDroid) [inline assets/src/edit-story/app/media/local/useContextValueProvider.js:81]: Can we add a todo / comment that this is just temporary and needs to be changed?

Comment (cvolzke4) [reply assets/src/edit-story/app/media/local/useContextValueProvider.js:81]: Today, the page token for local uploaded media is the page number.
What change did you have in mind?

Comment (winterDroid) [reply assets/src/edit-story/app/media/local/useContextValueProvider.js:81]: Ah so for the local it will always stay like that. Then it's fine.

**Comment by csossi** [comment]
Verified in QA

**Pull request merged**